sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java

```diff
@@ -292,38 +292,46 @@ private void saveIssues(SensorContext sensorContext, Issue[] issues, TypeScriptR
     for (Issue issue : issues) {
       InputFile inputFile = fs.inputFile(fs.predicates().hasAbsolutePath(issue.name));
       if (inputFile != null) {
-        RuleKey ruleKey = typeScriptRules.ruleKeyFromTsLintKey(issue.ruleName);
-        NewIssue newIssue = sensorContext.newIssue().forRule(ruleKey);
-        NewIssueLocation location = newIssue.newLocation();
-        location.on(inputFile);
-        location.message(issue.failure);
-
-        // semicolon rule
-        if (ruleKey.rule().equals("S1438")) {
-          location.at(inputFile.selectLine(issue.startPosition.line + 1));
-
-        } else if (!TypeScriptRules.FILE_LEVEL_RULES.contains(ruleKey.rule())) {
-          location.at(inputFile.newRange(
-            issue.startPosition.line + 1,
-            issue.startPosition.character,
-            issue.endPosition.line + 1,
-            issue.endPosition.character));
-        }
+        saveIssue(sensorContext, typeScriptRules, issue, inputFile);
+      }
+    }
+  }
 
-        newIssue.at(location);
+  private void saveIssue(SensorContext sensorContext, TypeScriptRules typeScriptRules, Issue issue, InputFile inputFile) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "saveIssue" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-165&issues=AWC7YjLnqDrCo5ysghE3&open=AWC7YjLnqDrCo5ysghE3)

[ISSUE_KEY]: AWC7YjLnqDrCo5ysghE3